pyHS100/smartstrip.py

```diff
@@ -0,0 +1,190 @@
+import logging
+from typing import Any, Dict
+
+from pyHS100 import SmartPlug
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SmartStrip(SmartPlug):
+    """Representation of a TP-Link Smart Power Strip.
+
+    Usage example when used as library:
+    p = SmartStrip("192.168.1.105")
+    # print the devices alias
+    print(p.alias)
+    # change state of plug
+    p.state = "ON"
+    p.state = "OFF"
+    # query and print current state of plug
+    print(p.state)
+
+    Errors reported by the device are raised as SmartDeviceExceptions,
+    and should be handled by the user of the library.
+
+    Note:
+    The library references the same structure as defined for the D-Link Switch
+    """
+
+    def __init__(self,
+                 host: str,
+                 protocol: 'TPLinkSmartHomeProtocol' = None) -> None:
+        SmartPlug.__init__(self, host, protocol)
+        self.emeter_type = "emeter"
+        self.plugs = []
+        for plug in self.sys_info["children"]:
+            self.plugs.append(SmartPlug(host, protocol, context=plug["id"]))
+
+    @property
+    def state(self) -> Dict[int, str]:
+        """
+        Retrieve the switch state
+
+        :returns: list with the state of each child plug
+                  SWITCH_STATE_ON
+                  SWITCH_STATE_OFF
+                  SWITCH_STATE_UNKNOWN
+        :rtype: dict
+        """
+        states = {}
+        for plug in range(self.num_children):
+            relay_state = self.sys_info["children"][plug]["state"]
```

Comment:
I would still prefer if `self.plugs` were a dict instead of a list (makes also `if X in` comparisons nicer when checking if a plug exists, see my commentary below), making `for plug in self.plugs` is preferable here. Please move calling `self.sysinfo` before the loop to avoid unnecessary I/O (Generally speaking I think we should cache `sysinfo` and refresh the cache where necessary, but that's for another PR..).